docs/sdk/src/reference_docs/frame_currency.rs

```diff
@@ -1,8 +1,99 @@
 //! FRAME Currency Abstractions and Traits
 //!
-//! Notes:
+//! TODO:
 //!
-//! - History, `Currency` trait.
-//! - `Hold` and `Freeze` with diagram.
-//! - `HoldReason` and `FreezeReason`
-//! 	- This footgun: https://github.com/paritytech/polkadot-sdk/pull/1900#discussion_r1363783609
+//! - [x] History, `Currency` trait.
+//! - [ ] `Hold` and `Freeze` with diagram.
```

Comment:
not sure what kind of diagram would make it clear... maybe some bars diagram with Freeze and Hold balance overlapped?

Reply:
It should not only be holds and freeze overlapped, but also free, spendable, untouchable, see https://github.com/paritytech/polkadot-sdk/issues/1833#issuecomment-1805764506

However, I don't think it should live here... These are concepts related to the Balances pallet, so I think should live in the Balances pallet docs.